lib/Differentiator/DerivativeBuilder.cpp

```
@@ -179,6 +179,11 @@ namespace clad {
       return {};
     }
     FD = FD->getDefinition();
+    auto* Attr = FD->getAttr<clang::AnnotateAttr>();
+    // Skip functions marked with 'annotate("non_differentiable")'.
+    if (Attr && Attr->getAnnotation() == "non_differentiable") {
```

We should not just skip this. We should report an error, if a user has explicitly requested to differentiate a non-differentiable marked function.

---

ok

---

Can you please add a test case in which a user tries to request a derivative of a function (using `clad::differentiate`) that is explicitly marked as `non_differentiable`?